reactor-core/src/main/java/reactor/core/publisher/FluxRefCountGrace.java

```diff
@@ -63,6 +63,7 @@ public int getPrefetch() {
 	public Object scanUnsafe(Attr key) {
 		if (key == Attr.PREFETCH) return getPrefetch();
 		if (key == Attr.PARENT) return source;
+		if (key == Attr.RUN_STYLE) return Attr.RunStyle.ASYNC;
```

Comment:
actually, after looking carefully at the operator I think this one is `SYNC`: the timeout only is for delaying cancellation from source. The `RefCountInner` isn't scheduled on a separate thread. By the way said inner should be covered>

Reply:
Fixed in commit [26552f6](https://github.com/reactor/reactor-core/pull/2123/commits/26552f67f2cfd7c10e59e60f3b905504c55b6e1d)